apps/framework-cli/src/cli/routines/seed_data.rs

```diff
@@ -202,114 +558,289 @@ pub async fn seed_clickhouse_tables(
-            body.trim()
-                .parse::<usize>()
-                .map_err(|e| RoutineFailure::new(remote_count_failure(), e))?
-        };
-        let total_rows = match limit {
-            None => remote_total,
-            Some(l) => min(remote_total, l),
-        };
-        let order_by_clause = match order_by {
-            None => {
-                let table = infra_map.tables.get(&table_name).ok_or_else(|| {
-                    RoutineFailure::error(Message::new(
-                        "Seed".to_string(),
-                        format!("{table_name} not found."),
-                    ))
-                })?;
-                let fields = table
-                    .order_by
-                    .iter()
-                    .map(|field| format!("`{field}` DESC"))
-                    .join(", ");
-                if !fields.is_empty() {
-                    format!("ORDER BY {fields}")
-                } else if total_rows <= batch_size {
-                    "".to_string()
-                } else {
-                    return Err(RoutineFailure::error(Message::new(
-                        "Seed".to_string(),
-                        format!("Table {table_name} without ORDER BY. Supply ordering with --order-by to prevent the same row fetched in multiple batches."),
-                    )));
-                }
+    // Get available remote tables for validation (unless specific table is requested)
+    let remote_tables = if table_name.is_some() {
+        // Skip validation if user specified a specific table
+        None
+    } else {
+        match get_remote_tables(local_clickhouse, remote_config).await {
+            Ok(tables) => Some(tables),
+            Err(e) => {
+                warn!("Failed to query remote tables for validation: {:?}", e);
+                display::show_message_wrapper(
+                    MessageType::Info,
+                    Message::new(
+                        "Validation".to_string(),
+                        "Skipping table validation - proceeding with seeding".to_string(),
+                    ),
+                );
+                None
             }
-            Some(order_by) => format!("ORDER BY {order_by}"),
-        };
-        let mut i: usize = 0;
-        'table_batches: while copied_total < total_rows {
-            i += 1;
-            let limit = match limit {
-                None => batch_size,
-                Some(l) => min(l - copied_total, batch_size),
-            };
+        }
+    };
 
-            let sql = format!(
-                "INSERT INTO `{local_db}`.`{table_name}` SELECT * FROM remoteSecure('{remote_host_and_port}', '{remote_db}', '{table_name}', '{remote_user}', '{remote_password}') {order_by_clause} LIMIT {limit} OFFSET {copied_total}"
+    // Process each table
+    for table_name in tables {
+        // Check if table should be skipped due to validation
+        if should_skip_table(&table_name, &remote_tables) {
```

Comment:
just write is_some_and instead of a new function?

Reply:
I think that's ok to encapsulate

Reply:
hm i don't get find that function more readable but it's fine i guess